consensus-types/types/attestation_data.go

```diff
@@ -79,9 +79,10 @@ func (a *AttestationData) MarshalSSZ() ([]byte, error) {
 	return buf, ssz.EncodeToBytes(buf, a)
 }
 
-// UnmarshalSSZ unmarshals the AttestationData object from SSZ format.
-func (a *AttestationData) UnmarshalSSZ(buf []byte) error {
-	return ssz.DecodeFromBytes(buf, a)
+// NewFromSSZ creates a new AttestationData object from SSZ format.
+func (*AttestationData) NewFromSSZ(buf []byte) (*AttestationData, error) {
+	a := &AttestationData{}
+	return a, ssz.DecodeFromBytes(buf, a)
```

Comment:
this makes an extra copy of `AttestationData` and does not initialize the pointer receiver.
Can you explain why we changed this?

Reply:
So actually this is what is done in others NewFromSSZ methods, which we didn't modify.
It should be fine, but I will recheck to see if there are any unnecessary copies

Reply:
With the latest approach in this PR, no new memory is created unnecessarily